File: micronaut-mcp-server-java-sdk/src/test/java/io/micronaut/mcp/server/stateless/sync/prompts/Prompts.java
package io.micronaut.mcp.server.stateless.sync.prompts;
/*
//tag::fakepackage[]
package example.micronaut;

//end::fakepackage[]
*/
import io.micronaut.context.annotation.Requires;
//tag::imports[]
import io.micronaut.mcp.annotations.Prompt;
import io.micronaut.mcp.annotations.PromptArg;
import jakarta.inject.Singleton;

//end::imports[]

@Requires(property = "spec.name", value = "StatelessSyncPromptsAnnotationsTest")
//tag::clazz[]
@Singleton
class Prompts {
    /**
     *
     * @return Chess statistics
     */
    @Prompt(name = "chess-statistics", description = "Displays statistics for chess games")
    String prompt(@PromptArg(description = "Player Name") String name) {
        return String.format("You generate chess statistics for %s ....", name);
    }
}
//end::clazz[]

File: test-suite-docs-kotlin/src/test/kotlin/io/micronaut/mcp/docs/prompts/Prompts.kt
package io.micronaut.mcp.docs.prompts
/*
//tag::fakepackage[]
package example.micronaut

//end::fakepackage[]
*/
import io.micronaut.context.annotation.Requires
//tag::imports[]
import io.micronaut.mcp.annotations.Prompt
import io.micronaut.mcp.annotations.PromptArg
import jakarta.inject.Singleton

//end::imports[]

@Requires(property = "spec.name", value = "PromptsTest")
//tag::clazz[]
@Singleton
class Prompts {
    /**
     *
     * @return Chess statistics
     */
    @Prompt(name = "chess-statistics", description = "Displays statistics for chess games")
    fun prompt(@PromptArg(description = "Player Name") name: String): String {
        return String.format("You generate chess statistics for %s ....", name)
    }
}
//end::clazz[]
